src/spatial_ref/srs.rs

```diff
@@ -33,6 +33,61 @@ impl CoordTransform {
         })
     }
 
+    /// Transform bounding box, densifying the edges to account for nonlinear
+    /// transformations.
+    ///
+    /// # Arguments
+    /// * bounds - array of [axis0_min, axis1_min, axis0_max, axis1_min],
+    ///            interpreted in the axis order of the source SpatialRef,
+    ///            typically [xmin, ymin, xmax, ymax]
+    /// * densify_pts - number of points per edge (recommended: 21)
+    ///
+    /// # Returns
+    /// Some([f64; 4]) with bounds in axis order of target SpatialRef
+    /// None if there is an error.
+    #[cfg(all(major_ge_3, minor_ge_4))]
+    pub fn transform_bounds(&self, bounds: &[f64; 4], densify_pts: i32) -> Result<([f64; 4])> {
+        let mut out_xmin: f64 = 0.;
+        let mut out_ymin: f64 = 0.;
+        let mut out_xmax: f64 = 0.;
+        let mut out_ymax: f64 = 0.;
+
+        let ret_val = unsafe {
+            gdal_sys::OCTTransformBounds(
+                self.inner,
+                bounds[0] as c_double,
+                bounds[1] as c_double,
+                bounds[2] as c_double,
+                bounds[3] as c_double,
+                &mut out_xmin as *mut c_double,
+                &mut out_ymin as *mut c_double,
+                &mut out_xmax as *mut c_double,
+                &mut out_ymax as *mut c_double,
+                densify_pts as c_int,
+            ) == 1
+        };
+
+        if ret_val {
+            Ok([out_xmin, out_ymin, out_xmax, out_ymax])
+        } else {
+            let err = _last_cpl_err(CPLErr::CE_Failure);
+            let msg = if let GdalError::CplError { msg, .. } = err {
```

Comment:
A `match` might look nicer here (you can probably drop the `trim`), but it probably doesn't matter too much.

Reply:
This was copy-pasted from `transform_coords` for consistency; ideally both would get updated to be the same if `match` is more appropriate.  I can update just this one here and the other in a follow up PR, or both in a follow up PR.

I'd probably also want to invert the logic a little bit so that only the error case is in the conditional and have a bare `Ok`:

```
if !ret_val {...handle error and return... }

Ok([out_xmin, out_ymin, out_xmax, out_ymax])
```

But didn't do so in this PR out of consistency.

Reply:
>  I can update just this one here and the other in a follow up PR, or both in a follow up PR.

Right, it's fine either way.